### DISCUSSION TITLE
프로젝트의 디렉토리 구조

### DISCUSSION THREAD

**Discussion by woowapark** [woowacourse/react-lotto#40] [토론해보기]
- 디렉토리 구조를 정하는 기준에는 무엇이 있을까요?
- 이번 프로젝트에서 어떤 구조를 사용했나요?
    - 왜 그런 구조로 구분하였나요?
    - 다른 방법으로 구성해본다면 어떻게 해볼 수 있을까요?

**Comment by hchayan** [comment]
사용하는것끼리 가까운데 위치하는게 최고...!

**Reply by zereight** [reply]
그럼 App.js에 모두 때려박는다?

**Reply by zigsong** [reply]
대박

**Reply by ddongule** [reply]
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

**Reply by hchayan** [reply]
> 그럼 App.js에 모두 때려박는다?

아니 나누긴 나눠야죠 ㅋㅋㅋㅋ

**Reply by wmakerjun** [reply]
TPO에 맞춰서

**Reply by igy95** [reply]
거의 CGV 패턴

**Reply by 0307kwon** [reply]
app 주도 개발ㅋㅋㅋㅋㅋ

**Comment by jho2301** [comment]
components디렉토리에서 조립할수있는 컴포넌트들을 만들고
pages에서 하나의 페이지에서 필요한 컴포넌트들을 조합한다

**Reply by jho2301** [reply]
유지보수의 용이성이 가장 중요한 것 같다. 
나중에 다시 돌아와서 기능을 추가,수정해도 직관적으로, 빨리, 최소한으로 코드를 짤 수 있는것

**Comment by sunhpark42** [comment]
파일이 많아졌을 때, 코드 변경이 필요할 시 길을 빠르게 찾아가기 위해서.

**Comment by seojihwan** [comment]
재 사용가능한 컴포넌트들을 별도의 폴더에 배치
Page단위의 컴포넌트들을 별도의 폴더로 배치

상태를 가지는 컴포넌트와 Presentational을 담당하는 컴포넌트로의 구조도 사용가능하다고 생각합니다.

**Comment by SunYoungKwon** [comment]
이번 미션은 이렇게 했어요
- src
  - components
    - 각 컴포넌트
    - pages
    - shared : 이 앱에서만 사용하지만 앱 내 여러 곳에서 재사용하고 있는 컴포넌트
    - common : 이 앱이 아니더라도 범용적으로 사용할 수 있는 컴포넌트
  - redux
  - constants
